### DISCUSSION TITLE
Annotation do not support clickable urls

### DISCUSSION THREAD

**Discussion by valeriast** [community/community#72821] [Actions]
When I try to create an annotation passing an external url in the message property of the annotation, GitHub doesn't make it clickable only if the url is part of GitHub domain which is not my case. 

Is there a way to make external urls clickable the same way as the internal ones?

**Comment by ktauchathuranga** [comment]
refer this one
[GitHub Links](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links)

**Comment by Mr-Gleidson** [comment]
GitHub doesn't automatically make external URLs in annotations clickable, and this behavior is intentional to prevent potential security risks associated with arbitrary external links.

If you want to make external URLs in your GitHub annotations clickable, you might need to take a different approach. One way to achieve this is by using HTML to create a clickable link within your annotation message.

Here's an example of how you can create a clickable external link within your annotation message:

```
## My Annotation

You can visit [example.com](https://www.example.com) for more information.

```
By using standard Markdown syntax, you can include clickable external links in your annotation. This approach ensures that users can click on the link to access the external URL. Remember to use full URLs, including the "https://" or "http://" part.

Please keep in mind that this won't change GitHub's default behavior but provides a readable and functional solution for your specific use case.

**Comment by valeriast** [comment]
I believe I may be doing something wrong. I have tried using this format but it still showing as a text. I'm using ocktokit in javascript to create the annotations.

This is the annotation schema:
![image](https://github.com/community/community/assets/28829768/dfef9033-3130-4188-a7e0-37d9bb05fd21)

This is the outcome:
![image](https://github.com/community/community/assets/28829768/6fefc723-8c18-4b39-843c-b60b2ceca956)

**Reply by cmmanu** [reply]
Have you found a solution yet?
I have the same problem. Neither markdown or html work.

**Reply by valeriast** [reply]
Last time I checked it was not supported so I just applied the url as a string which honestly is not ideal

**Comment by xethorn** [comment]
To add to the above, we use annotations for things like linting and reporting back on failed tests. Being able to use full markdown expressions would allow us to provide a richer experience to our developers – not just to point to the contribution guidelines, but also to print out test inputs and outputs. The alternative would be to use a separate user with the right permissions that adds comments. It would work, but it's definitely not as well integrated.
